python/ray/serve/_private/router.py

```diff
@@ -695,12 +699,56 @@ def assign_request(
         *request_args,
         **request_kwargs,
     ) -> concurrent.futures.Future[ReplicaResult]:
-        return asyncio.run_coroutine_threadsafe(
+        """Schedules assign_request call on the internal asyncio loop.
+
+        This method uses `run_coroutine_threadsafe` to execute the actual request
+        assignment logic (`_asyncio_router.assign_request`) on the dedicated
+        asyncio event loop thread. It returns a `concurrent.futures.Future` that
+        can be awaited or queried from the calling (potentially synchronous) thread.
```

Comment:
```suggestion
        can be awaited or queried from the calling thread.
```
"synchronous thread" isn't really a meaningful concept and the fact it's a concurrent futures future already communicates the semantics